app/controllers/api/v1/users_controller.rb

```diff
@@ -4,6 +4,7 @@ class UsersController < ApiController
       include Api::UsersController
 
       before_action :authenticate!
+      before_action :authorize_super_admin, only: [:unpublish]
```

Comment:
Reading Notes: This is a method in the v0 policy ecosystem.

Editorialization: My preference is that we have policy objects handling this behavior and rely on `authorize(resource)` however that is a larger refactor.

By continuing to use the `authorize_super_admin` we're creating circumstances that mean we have duplication of policy knowledge/enforcement.  This isn't quite a blocker, but instead a "musing outlaid"

Reply:
Oh this makes a lot of sense!

> My preference is that we have policy objects handling this behavior and rely on authorize(resource) however that is a larger refactor.

If I understand this correctly, the existing `user_policy` could be used to replace this with an `unpublish` or `admin_unpublish` method that validates the user attempting this action is a super admin.

I can push these changes today 👍🏼

Reply:
I had the same question/concern in the Suspend User API endpoint PR: https://github.com/forem/forem/pull/18043/files#r918138940

Reply:
Whoops, yeah makes sense I overlooked this in both PRs. I'm pushing the proper validation using the policy in both of them soon. Thanks for pointing this out @joshpuetz  and @jeremyf  🙏🏼